Regression in 0.8.0: build fails due to unconditional usage of `aligned_alloc`: `memoryManagement.c:48:28: error: implicit declaration of function 'aligned_alloc'`

```
/opt/local/var/macports/build/idris2-b99237ae/work/compwrap/cc/opt/local/bin/gcc-mp-14 -Os -arch ppc -Wall -fPIC -Wall -fPIC -Wall -fPIC -O2 -I/opt/local/include  -c -o memoryManagement.o memoryManagement.c
memoryManagement.c: In function 'idris2_newValue':
memoryManagement.c:48:28: error: implicit declaration of function 'aligned_alloc' [-Wimplicit-function-declaration]
   48 |   Value *retVal = (Value *)aligned_alloc(
      |                            ^~~~~~~~~~~~~
memoryManagement.c:6:1: note: include '<stdlib.h>' or provide a declaration of 'aligned_alloc'
    5 | #include "runtime.h"
  +++ |+#include <stdlib.h>
    6 | 
memoryManagement.c:48:28: warning: incompatible implicit declaration of built-in function 'aligned_alloc' [-Wbuiltin-declaration-mismatch]
   48 |   Value *retVal = (Value *)aligned_alloc(
      |                            ^~~~~~~~~~~~~
memoryManagement.c:48:28: note: include '<stdlib.h>' or provide a declaration of 'aligned_alloc'
make[2]: *** [memoryManagement.o] Error 1
make[2]: Leaving directory `/opt/local/var/macports/build/idris2-b99237ae/work/Idris2-0.8.0/support/refc'
make[1]: *** [build-refc] Error 2
```

`aligned_alloc` is not guaranteed to be available and should be checked at configure time.

--- a/support/refc/memoryManagement.c
+++ b/support/refc/memoryManagement.c
@@ -42,22 +42,25 @@ void idris2_dumpMemoryStats(void) {
 void idris2_dumpMemoryStats() {}
 #endif
 
-Value *idris2_newValue(size_t size)
-{
+Value *idris2_newValue(size_t size) {
   /* Try to get memory aligned to pointer-size. Prefer C11 aligned_alloc
-    (not available on some platforms like older macOS), then posix_memalign,
-    and finally fall back to malloc which typically returns pointer-aligned
-    memory suitable for our needs. */
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) && \
+     (not available on some platforms like older macOS), then posix_memalign,
+     and finally fall back to malloc which typically returns pointer-aligned
+     memory suitable for our needs. */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) &&              \
     !defined(__APPLE__) && !defined(_WIN32)
-  Value *retVal = (Value *)aligned_alloc(sizeof(void *),
-                                        ((size + sizeof(void *) - 1) / sizeof(void *)) * sizeof(void *));
-#elif ( (defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 200112L)) || \
-        (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 600))            ) && \
+  Value *retVal = (Value *)aligned_alloc(
+      sizeof(void *),
+      ((size + sizeof(void *) - 1) / sizeof(void *)) * sizeof(void *));
+#elif ((defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 200112L)) ||           \
+       (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 600))) &&                  \
     !defined(_WIN32)
   Value *retVal = NULL;
-  IDRIS2_REFC_VERIFY(posix_memalign((void**)&retVal, sizeof(void *), 
-        ((size + sizeof(void *) - 1) / sizeof(void *)) * sizeof(void *)) == 0, "posix_memalign failed");
+  IDRIS2_REFC_VERIFY(
+      posix_memalign((void **)&retVal, sizeof(void *),
+                     ((size + sizeof(void *) - 1) / sizeof(void *)) *
+                         sizeof(void *)) == 0,
+      "posix_memalign failed");
 #else
   Value *retVal = (Value *)malloc(size);
 #endif

